https://issues.redhat.com/browse/ACM-6946 Observability doc updates

dockerymick opened the pull request:
https://issues.redhat.com/browse/ACM-6946

dockerymick committed:
Observability doc updates

dockerymick committed:
updates

dockerymick committed:
updates

dockerymick committed:
Improving the doc

dockerymick committed:
More improvements

dockerymick committed:
updates

dockerymick committed:
Updates

dockerymick committed:
Updates

dockerymick committed:
adding hidden comments from older PR

dockerymick committed:
adding change for 2.9 from ACM-5462

dockerymick committed:
More updates

dockerymick committed:
Update observability_arch.adoc

live review

dockerymick commented on observability/observability_enable.adoc:
```diff
@@ -1,7 +1,9 @@
+[#managing-observability-service]
+= Managing the observability service
+
+When you enable the observability component, you can use metrics and configure dashboards to make cluster resource information visible, help you save cost, and prevent unnecessary events.
```
```suggestion
When you enable the observability component, you can use metrics and configure Grafana dashboards to make cluster resource information visible, help you save cost, and prevent unnecessary events.
```

dockerymick committed:
Update observability/observability_enable.adoc

dockerymick commented on observability/observability_enable.adoc:
```diff
@@ -436,19 +440,35 @@ oc apply -f multiclusterobservability_cr.yaml
++
+[source,bash]
+----
+need command here
```
@dislbenn will get command for this

dislbenn replied:
This command will get the `multicluster-observability-operator` deployment  and display the labels associated with the resource:
```bash
$ oc get deploy multicluster-observability-operator -n open-cluster-management --show-labels

NAME                                  READY   UP-TO-DATE   AVAILABLE   AGE   LABELS
multicluster-observability-operator   1/1     1            1           35m   installer.name=multiclusterhub,installer.namespace=open-cluster-management

```

dockerymick commented on release_notes/whats_new.adoc:
```diff
@@ -22,6 +22,9 @@
 
+//Mikela added this entry; need to confirm that this is correct | 08/28/23
+//2.9:ACM-6677
+* The observability component is deployed by the `multiclusterhub-operator`. 
```
Mikela - add link to the related topic (intro)

dockerymick replied:
```suggestion
* The observability component is deployed by the `multiclusterhub-operator`. See link:../observability/observability_arch.adoc#observability-arch[Observability architecture] for more details.
```

dockerymick commented on observability/observe_environments.adoc:
```diff
@@ -284,43 +253,16 @@ Observability components require 2701mCPU and 11972Mi memory to install the obse
 | 3072
 |===
 
-[#persistent-stores-observability]
```
add the acm_managed_cluster_labels to the metric type table @dockerymick 

@dislbenn will help get this info

Gauge, no label/task, Stable metric

dockerymick commented on observability/observability_enable.adoc:
```diff
@@ -436,19 +440,35 @@ oc apply -f multiclusterobservability_cr.yaml
++
+[source,bash]
+----
+need command here
```
Adding suggested command

```suggestion
oc get deploy multicluster-observability-operator -n open-cluster-management --show-labels

NAME                                  READY   UP-TO-DATE   AVAILABLE   AGE   LABELS
multicluster-observability-operator   1/1     1            1           35m   installer.name=multiclusterhub,installer.namespace=open-cluster-management
```

dockerymick commented on observability/observability_enable.adoc:
```diff
@@ -436,19 +440,35 @@ oc apply -f multiclusterobservability_cr.yaml
 +
 *Note:* If you want to exclude specific managed clusters from collecting the observability data, add the following cluster label to your clusters: `observability: disabled`.
 
+. Verify that the `multicluster-observability-operator` pod is being deployed by the `multiclusterhub-operator` deployment by viewing the `labels` section of the `multicluster-observability-operator` deployment. Run the following command to view the deployment:
```
```suggestion
. Verify that the `multicluster-observability-operator` pod is being deployed by the `multiclusterhub-operator` deployment by viewing the `labels` section of the `multicluster-observability-operator` deployment. Run the following command to get the `multicluster-observability-operator` deployment and display the labels associated with the resource:
```

dockerymick committed:
Apply suggestions from code review

dockerymick committed:
Apply suggestions from code review

dockerymick commented on observability/observe_environments_intro.adoc:
```diff
@@ -1,8 +1,28 @@
 
-* xref:../observability/observe_environments.adoc#observing-environments[Observing environments]
+The observability component is a service you can use to understand the state of your clusters. The observability service is deployed by the `multiclusterhub-operator` pod, which is complete during the {product-title-short} installation. By default, observability is enabled on your hub cluster.
+//does it make sense to have the support section here? 
```
```suggestion
//does it make sense to have the support section here? 

```

dockerymick committed:
Update observability/observe_environments_intro.adoc

dislbenn reviewed (approved):
Looks good to me. Thanks @dockerymick!

jc-berger commented on observability/docinfo.xml:
```diff
@@ -2,7 +2,7 @@
 <subtitle>Observability</subtitle>
 <abstract>
-    <para>Read more to learn how to optimize your managed clusters by enabling and customizing the observability service.</para>
+    <para>With the observability component, you can gain insight about your managed clusters. Observability is available through the {product-title}.</para>
```
IN RHACM, do we default to action oriented in most cases? 

If so, the second sentence here could be:

Access observability through the {product title}

Note, I don't think you need to make this change. The sentence is good as is :) I know some products defer to action oriented in most cases, hence, thought I'd suggest it.

dockerymick replied:
@jc-berger great question and suggestion, we do default to action oriented descriptions. we have been working to improve the descriptions in our `docinfo.xml` files throughout the repo (least priority). I I think i should delete the second statement bc it is redundant. I think it's apparent that the user can access the feature through RHACM.

jc-berger commented on observability/observability_arch.adoc:
```diff
@@ -0,0 +1,66 @@
+[#observability-arch]
+= Observability architecture
+
+The observability service can be used to get a status and metrics for your hub cluster and managed clusters. When you enable observability, the `multicluster-observability-operator` pod is deployed by the `multiclusterhub-operator` automatically to each imported or created hub cluster.
```
In this case, since we're already calling out the user specifically (your hub cluster), I'd make the entire sentence user focused/action oriented. 

Use the observability service to get a status and metrics for your hub cluster and managed clusters.

dockerymick replied:
Great point!

jc-berger commented on observability/observability_arch.adoc:
```diff
@@ -0,0 +1,66 @@
+//is the endpoint operator the same as the observaility add-on controller? 
+//When the service is enabled, the `observability-endpoint-operator` is automatically deployed to each imported or created cluster. This controller collects the data from {ocp} Prometheus, then sends it to the {product-title-short} hub cluster. If the hub cluster imports itself as the `local-cluster`, observability is also enabled on it and metrics are collected from the hub cluster.
+
+//original
```
I assume the content in these three paragraphs are the hidden comments, hence, not needing review because this content is not being published?

dockerymick replied:
Correct

jc-berger commented on observability/observability_arch.adoc:
```diff
@@ -0,0 +1,66 @@
+//need to update the diagram for 2.8 and 2.9, 
+//D suggested to add a box for Thanos on the hub cluster side + metrics-collector on the managed cluster side, get managed cluster metrics are collected and sent to the hub cluster | MJ | 08/30/23
+
+The following diagram shows the components of the observability component:
```
Minimalism: avoid redundancy. 

The following diagram shows the observability components:

dockerymick replied:
```suggestion
The following diagram shows the components of observability:
```

jc-berger commented on observability/observability_arch.adoc:
```diff
@@ -0,0 +1,66 @@
+
+image:../images/observability-arch-24.png[Multicluster observability architecture]
+
+*Note:* The _on-demand log_ provides access for engineers to get logs for a given pod in real-time. Logs from the hub cluster are not aggregated. These logs can be accessed with the search service and other parts of the console.
```
We've been using second person to refer to users throughout. Now, we're saying engineers. For consistency, can we use second person here? So something like:

*Note:* The _on-deand_ provides you access to get logs for your pods in real time. Logs from the hub cluster are not aggregated. To access the hub cluster logs, use the search service and other parts of the console.

dockerymick replied:
Incorporating suggestion after peer review
```suggestion
*Note:* The _on-demand log_ provides you access to get logs for your pods in real time. Logs from the hub cluster are not aggregated. To access the hub cluster logs, use the search service and other parts of the console.
```

jc-berger commented on observability/observability_arch.adoc:
```diff
@@ -0,0 +1,66 @@
+
+The components of the observability architecture include the following items:
+
+- The multicluster hub operator (`multiclusterhub-operator` pod) deploys the `multicluster-observability-operator` pod. It sends hub cluster data to your managed clusters.
```
IBM SG wants us to avoid parenthesis unless it's specifying an acronym. I think we can simply say:

The multicluster hub operator, or the `multicluterhub-operator` pod, deploys the `multicluster-observability-operator` pod.

dockerymick replied:
Incorporating suggestion after peer review
```suggestion
- The multicluster hub operator, also known as the `multiclusterhub-operator` pod, deploys the `multicluster-observability-operator` pod. It sends hub cluster data to your managed clusters.
```

jc-berger commented on observability/observability_enable.adoc:
```diff
@@ -436,19 +440,38 @@ oc apply -f multiclusterobservability_cr.yaml
 All the pods in `open-cluster-management-observability` namespace for Thanos, Grafana and Alertmanager are created. All the managed clusters connected to the {product-title-short} hub cluster are enabled to send metrics back to the {product-title-short} Observability service.
 
-. Validate that the observability service is enabled and the data is populated by launching the Grafana dashboards. Click the **Grafana link** that is near the console header, from either the console _Overview_ page or the _Clusters_ page.
+. Validate that the observability service is enabled and the data is populated by launching the Grafana dashboards. Click the *Grafana link* that is near the console header, from either the console _Overview_ page or the _Clusters_ page.
```
Since there are two separate actions, I would make them two separate steps:

1. Validate that the observaiblity service is enabled and the data is populated by launching the Grafana dashboards. 
2. Click the *Grafana link* that is near the console header, from either the console _Overview_ page or the _Clusters_ page.

dockerymick replied:
Incorporating suggestion after peer review
```suggestion
. Validate that the observability service is enabled and the data is populated by launching the Grafana dashboards. 

. Click the *Grafana link* that is near the console header, from either the console _Overview_ page or the _Clusters_ page.
```

jc-berger commented on observability/observability_enable.adoc:
```diff
@@ -436,19 +440,38 @@ oc apply -f multiclusterobservability_cr.yaml
-. Validate that the observability service is enabled and the data is populated by launching the Grafana dashboards. Click the **Grafana link** that is near the console header, from either the console _Overview_ page or the _Clusters_ page.
+. Validate that the observability service is enabled and the data is populated by launching the Grafana dashboards. Click the *Grafana link* that is near the console header, from either the console _Overview_ page or the _Clusters_ page.
++
+You can also access the {ocp-short} 3.11 Grafana dashboards with the following URL: `https://$ACM_URL/grafana/dashboards`. Select the folder named _OCP 3.11_ to view the {ocp-short} 3.11 dashboards.
```
Since this is part of the procedure, I'd make it a step. Again, split it up into multiple steps, since there are multiple actions. 

So the whole thing would be: 

.Procedure

1. Validate that the observaiblity service is enabled and the data is populated by launching the Grafana dashboards.
2. Click the Grafana link that is near the console header, from either the console Overview page or the Clusters page.
3. Access the {ocp-short} 3.11 Grafana dashboards with the following URL: `https://$ACM_URL/grafana/dashboards`. 
4. To view the {ocp-short} dashboards, select the folder named _OCP 3.11_ . 
5. To verify that the `multicluster-observability-operator` pod is being deployed by the `multiclusterhub-operator` deployment, view the `labels` section of the `multicluster-observability-operator` deployment.
6. To get the `multicluster-observability-operator` deployment and display the labels associated with the resource, run the following command: 

< provide command > 

7. *Optional:* If you want to exclude specific managed clusters from collecting the observability data, add the following cluster label to your clusters: `observability: disabled`.

If I am understanding correctly, the note is more of an optional step, and I think it'd make a better flow if it were given after the command.

jc-berger commented on observability/observability_enable.adoc:
```diff
@@ -436,19 +440,38 @@ oc apply -f multiclusterobservability_cr.yaml
 +
 *Note:* If you want to exclude specific managed clusters from collecting the observability data, add the following cluster label to your clusters: `observability: disabled`.
 
+. Verify that the `multicluster-observability-operator` pod is being deployed by the `multiclusterhub-operator` deployment by viewing the `labels` section of the `multicluster-observability-operator` deployment. Run the following command to get the `multicluster-observability-operator` deployment and display the labels associated with the resource:
```
See my above comment for all the suggested revisions for the procedure steps.

jc-berger commented on observability/observe_environments_intro.adoc:
```diff
@@ -1,8 +1,29 @@
+[#observability-support]
+== Support
+
+- {product-title-short} is tested with and fully supported by Red Hat OpenShift Data Foundation (formerly Red Hat OpenShift Container Storage). 
```
IBM SG recommends avoiding parenthesis. I think it reads just as well to simply write:

{product-title-short} is tested with and fully supported by Red Hat OpenShift Data Foundation, formerly Red Hat OpenShift Container Storage.

dockerymick commented on observability/observability_arch.adoc:
```diff
@@ -0,0 +1,66 @@
+[#observability-arch]
+= Observability architecture
+
+The observability service can be used to get a status and metrics for your hub cluster and managed clusters. When you enable observability, the `multicluster-observability-operator` pod is deployed by the `multiclusterhub-operator` automatically to each imported or created hub cluster.
```
incorporating suggestion after peer review
```suggestion
Use the observability service to get a status and metrics for your hub cluster and managed clusters. When you enable observability, the `multicluster-observability-operator` pod is deployed by the `multiclusterhub-operator` automatically to each imported or created hub cluster.
```

dockerymick commented on observability/docinfo.xml:
```diff
@@ -2,7 +2,7 @@
 <subtitle>Observability</subtitle>
 <abstract>
-    <para>Read more to learn how to optimize your managed clusters by enabling and customizing the observability service.</para>
+    <para>With the observability component, you can gain insight about your managed clusters. Observability is available through the {product-title}.</para>
```
```suggestion
    <para>Enable the observability component to gain insight about your managed clusters. </para>
```

dockerymick committed:
Apply suggestions from code review

committing suggestions

dockerymick committed:
Update observability_enable.adoc

Updates after peer review

dockerymick committed:
Update observe_environments_intro.adoc

removing hidden comment bc it was confirmed during live review with Disiah. Also removed parenthesis

dockerymick commented on release_notes/whats_new.adoc:
```diff
@@ -22,6 +22,9 @@
 == Web console
 //team said nothing new this release, if that is the case, we can remove any section with no updates
 
+//Mikela added this entry; need to confirm that this is correct | 08/28/23
```
```suggestion

```

dockerymick committed:
Update release_notes/whats_new.adoc

dockerymick commented:
@jc-berger thank you so much for your review and feedback! I made some changes based on your suggestions. When you have an opportunity, will you re-review?

dockerymick committed:
Update observability_enable.adoc

jc-berger reviewed (approved):
/lgtm

dockerymick commented on observability/observability_enable.adoc:
```diff
@@ -436,9 +440,33 @@ oc apply -f multiclusterobservability_cr.yaml
+
+NAME                                  READY   UP-TO-DATE   AVAILABLE   AGE   LABELS
+multicluster-observability-operator   1/1     1            1           35m   installer.name=multiclusterhub,installer.namespace=open-cluster-management
+----
```
```suggestion
----

```

dockerymick committed:
Update observability/observability_enable.adoc

dockerymick commented:
@jc-berger thanks for your approval! Merging this PR
